src/TestSessionEnvironment.php

```diff
@@ -252,26 +258,14 @@ protected function moveRecursive($src, $dest, $ignore = [])
     }
 
     /**
-     * Assumes the database has already been created in startTestSession(), as this method can be called from
-     * _config.php where we don't yet have a DB connection.
-     *
-     * Persists the state to the filesystem.
-     *
-     * You can extend this by creating an Extension object and implementing either onBeforeApplyState() or
-     * onAfterApplyState() to add your own test state handling in.
+     * takes a state object (stdClass) and merges the existing state from getState()
+     * NB new state will not be overwritten
      *
-     * @param mixed $state
-     * @throws LogicException
-     * @throws InvalidArgumentException
+     * @param stdClass $state
+     * @return void
      */
-    public function applyState($state)
+    public function mergeWithExistingState($state)
```

Comment:
Yeah, I guess we'll have to leave invocation for this method within the `applyState` for now to keep backward compatibility. Otherwise, it will be a breaking change, since `applyState` is a public method.

Reply:
Do we need this as a public API here? Let's remove it?